algorithm/cgal/cast_util.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include "arrangement_util.h"
+#include "polygon_util.h"
+
+void cast_mesh_to_gps(const CGAL::Surface_mesh<EK::Point_3>& mesh,
+                      const EK::Plane_3& reference_plane,
+                      const EK::Vector_3& reference_vector,
+                      General_polygon_set_2& gps) {
+  CGAL::Surface_mesh<EK::Point_3> projected_mesh(mesh);
+  const auto& input_map = mesh.points();
+  auto& output_map = projected_mesh.points();
+  // Squash the mesh.
+  for (auto vertex : mesh.vertices()) {
+    const EK::Line_3 line(get(input_map, vertex),
+                          get(input_map, vertex) + reference_vector);
+    auto result = CGAL::intersection(line, reference_plane);
+    if (result) {
+      if (EK::Point_3* point = std::get_if<EK::Point_3>(&*result)) {
+        put(output_map, vertex, *point);
+      }
+    }
+  }
+  PlanarSurfaceMeshFacetsToPolygonSet(reference_plane, projected_mesh, gps);
+}
+
+void cast_mesh_to_polygons_with_holes(
+    const CGAL::Surface_mesh<EK::Point_3>& mesh,
+    const EK::Plane_3& reference_plane, const EK::Vector_3& reference_vector,
+    std::vector<CGAL::Polygon_with_holes_2<EK>>& simple_pwhs) {
+  General_polygon_set_2 gps;
+  std::vector<CGAL::Polygon_with_holes_2<EK>> pwhs;
+  cast_mesh_to_gps(mesh, reference_plane, reference_vector, gps);
+  copy_gps_to_pwhs(gps, pwhs);
+  simplifyPolygonsWithHoles(pwhs, simple_pwhs);
+}
+
+void cast_polygons_with_holes_to_polygons_with_holes(
```

Comment:
Why does this need to do anything?

Reply:
Ah, they end up projected onto a different plane, potentially.

Reply:
That makes sense :)